lib/types/beacon_chain/execution_payload.ex

```diff
@@ -50,4 +51,25 @@ defmodule Types.ExecutionPayload do
   def decode(%__MODULE__{} = map) do
     Map.update!(map, :base_fee_per_gas, &Ssz.decode_u256/1)
   end
+
+  @spec prepare_execution_payload(
```

Comment:
who calls this function?

Reply:
oh right!!! :sweat_smile: `prepare_execution_payload` is totally wrong 
So, I think that  `notify_forkchoice_updated` should be called inside the `on_block` handler :thinking: 

I going to check the prism consensus client: https://docs.prylabs.network/docs/how-prysm-works/optimistic-sync#22-the-engine-api. I hope that the go implementation is simpler than rust one :sweat_smile:

Reply:
I already removed the `prepare_execution_payload`.
But I am not sure where `notify_forkchoice_updated` should be called :disappointed:

Reply:
<img width="551" alt="Screenshot 2024-01-05 at 13 10 06" src="https://github.com/lambdaclass/lambda_ethereum_consensus/assets/1040971/c529ec3e-7d71-4bd0-8486-eee2b0d86d65">
this is in lighthouse. So everytime you compute a new head

Reply:
oh!! ok, I think I get it now. In our case, this call should be done on `ForkChoice.Helpers.get_head`, right?.

Reply:
we should ideally recompute a new head for after every processed block and then `get_head` should return the cached value. But yeah, for now we could add it there.